arbor/include/arbor/simulation.hpp

```diff
@@ -79,7 +81,7 @@ class ARB_ARBOR_API simulation {
 
     // Register a callback that will perform a export of the rank local
     // spike vector.
-    void set_local_spike_callback(spike_export_function = spike_export_function{});
+    void set_local_spike_callback(rank_spike_export_function = rank_spike_export_function{});
```

Comment:
This is a user supplied callback and thus is breaking API, which is currently not a showstopper since Arbor is <1.0 as such.
However, since the user will now be presented with duplicates of spikes (one for each rank we will transmit it to), we essentially expose internal details to them. 
That's problematic on a few levels (happy to discuss in detail, if you want)